From 5d1338e5a13bb1cb593afe638ceadf2c27c80bb3 Mon Sep 17 00:00:00 2001
From: Torsten Beyer <tb@pobox.com>
Date: Mon, 14 Sep 2026 17:11:28 +0000
Subject: [PATCH] Set minimum CPU voltage to 1.1V

The cubieboard2 dev spec says minimum voltage for lower frequencies should
be 1.0V and 1.1V respectively. Usage of cubieboards in OpenVario shows
unstable behaviour with voltages of 1.0V. Hence this change increases CPU
voltage to 1.1V for the lowest CPU speeds.

Rebased onto u-boot v2024.01, where operating-points is written as a list
of separate <kHz uV> entries rather than one enclosing block. The values
are unchanged from the version that applied to 2022.01.

Upstream-Status: Inappropriate [OpenVario specific]
---
 arch/arm/dts/sun7i-a20.dtsi | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/arch/arm/dts/sun7i-a20.dtsi b/arch/arm/dts/sun7i-a20.dtsi
index 55742996..cab4a0cd 100644
--- a/arch/arm/dts/sun7i-a20.dtsi
+++ b/arch/arm/dts/sun7i-a20.dtsi
@@ -106,15 +106,17 @@
 			reg = <0>;
 			clocks = <&ccu CLK_CPU>;
 			clock-latency = <244144>; /* 8 32k periods */
+			clock-frequency = <720000>;
 			operating-points =
 				/* kHz	  uV */
+				<1008000 1450000>,
 				<960000	1400000>,
 				<912000	1400000>,
 				<864000	1300000>,
 				<720000	1200000>,
 				<528000	1100000>,
-				<312000	1000000>,
-				<144000	1000000>;
+				<312000	1100000>,
+				<144000	1100000>;
 			#cooling-cells = <2>;
 		};
 
@@ -124,15 +126,17 @@
 			reg = <1>;
 			clocks = <&ccu CLK_CPU>;
 			clock-latency = <244144>; /* 8 32k periods */
+			clock-frequency = <720000>;
 			operating-points =
 				/* kHz	  uV */
+				<1008000 1450000>,
 				<960000	1400000>,
 				<912000	1400000>,
 				<864000	1300000>,
 				<720000	1200000>,
 				<528000	1100000>,
-				<312000	1000000>,
-				<144000	1000000>;
+				<312000	1100000>,
+				<144000	1100000>;
 			#cooling-cells = <2>;
 		};
 	};
